src/main/java/io/github/fablabsmc/fablabs/api/fluidvolume/v1/FluidHelper.java

```diff
@@ -0,0 +1,83 @@
+package io.github.fablabsmc.fablabs.api.fluidvolume.v1;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import io.github.fablabsmc.fablabs.api.fluidvolume.v1.containers.BlockFluidContainerProvider;
+import io.github.fablabsmc.fablabs.api.fluidvolume.v1.containers.EntityFluidContainerProvider;
+import io.github.fablabsmc.fablabs.api.fluidvolume.v1.world.WorldFluidCallback;
+import io.github.fablabsmc.fablabs.api.fluidvolume.v1.containers.ItemFluidContainerProvider;
+import io.github.fablabsmc.fablabs.api.fluidvolume.v1.volume.ImmutableFluidVolume;
+import io.github.fablabsmc.fablabs.api.fluidvolume.v1.volume.MultiFluidContainer;
+import io.github.fablabsmc.fablabs.api.fluidvolume.v1.volume.api.FluidContainer;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.entity.Entity;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Box;
+import net.minecraft.util.math.Direction;
+import net.minecraft.world.World;
+
+public final class FluidHelper {
+	/**
+	 * attempt to get a fluid container from the world from a specific block (or entities) for a given face.
+	 *
+	 * @param world the world
+	 * @param pos   the position
+	 * @param side  the side of the block/entity to access
+	 * @return ImmutableFluidVolume#EMPTY if no fluid container exists for the given side
+	 */
+	public static FluidContainer getContainer(World world, BlockPos pos, Direction side) {
+		BlockState state = world.getBlockState(pos);
+		Block block = state.getBlock();
+
+		FluidContainer container = ImmutableFluidVolume.EMPTY;
+
+		if (block instanceof BlockFluidContainerProvider) {
```

Comment:
Would move this into a default WorldFluidCallback.

Reply:
no, actually that reminds me I should optimize the callback

Reply:
why though?